plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/InferPodPathJob.java

```diff
@@ -0,0 +1,154 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.ui.server;
+
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.util.Hashtable;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.wst.server.core.ServerUtil;
+import org.jboss.tools.openshift.common.core.utils.StringUtils;
+import org.jboss.tools.openshift.core.connection.Connection;
+import org.jboss.tools.openshift.core.server.PodDeploymentPathProvider;
+
+import com.openshift.restclient.model.IService;
+
+/**
+ * 
+ * @author Viacheslav Kabanovich
+ *
+ */
+public class InferPodPathJob extends Job implements PropertyChangeListener {
+	static String POD_PATH_LOADING_MESSAGE = "loading...";
+
+	protected ServerSettingsWizardPageModel model;
+
+	protected IService service; //service to load pod path for. When path is loaded, service is set to null.
+	protected boolean isLoaded = false;
+	protected String customPodPath = null;
+	protected Hashtable<String, String> cache = new Hashtable<String, String>();
+	
+	public InferPodPathJob(ServerSettingsWizardPageModel model) {
+		super("Infer pod path");
+		this.model = model;
+	}
+
+	public boolean isLoaded() {
+		return isLoaded;
+	}
+
+	@Override
+	public boolean belongsTo(Object family) {
+		return ServerUtil.SERVER_JOB_FAMILY.equals(family);
+	}
+
+	@Override
+	protected IStatus run(IProgressMonitor monitor) {
+		if(service == null) {
+			return Status.OK_STATUS;
+		}
+		setLoadingMessage();
+		while(true) {
```

Comment:
There are 2 break conditions below. Cycle continues only while method loadPodPath() fails and stop() is not called.